writer.go

```diff
@@ -768,7 +768,7 @@ func (p *MediaPlaylist) SetDefaultKey(method, uri, iv, keyformat, keyformatversi
 	if keyformat != "" || keyformatversions != "" {
 		version(&p.ver, 5)
 	}
-	p.Key = &Key{method, uri, iv, keyformat, keyformatversions}
+	p.Key = &Key{Method: method, URI: uri, IV: iv, Keyformat: keyformat, Keyformatversions: keyformatversions}
```

Comment:
i think we'd need to set key ID here and add a param to the function?

Reply:
Wasn't sure if it'd matter. I'll add it